services/notification/internal/service/notification_service.go

```diff
@@ -3,24 +3,147 @@ package service
 import (
 	"context"
 	"fmt"
+	"log/slog"
+	"time"
+
+	"golang.org/x/sync/errgroup"
 
 	"github.com/samims/hcaas/services/notification/internal/model"
+	"github.com/samims/hcaas/services/notification/internal/store"
 )
 
+// NotificationService defines behavior for sending notifications
+// Implementations may enqueue or directly send notifications
 type NotificationService interface {
-	SendNotification(ctx context.Context, notification *model.Notification) error
+	// Start the background workers (for async delivery)
+	Start(ctx context.Context) error
+	// Send queues or sends a notification
+	Send(ctx context.Context, n *model.Notification) error
 }
 
+// notificationService is the default implementation of NotificationService
 type notificationService struct {
+	store       store.NotificationStorage
+	delivery    DeliveryService
+	workerLimit int
+	interval    time.Duration
+	l           *slog.Logger
 }
 
-func NewNotificationService() NotificationService {
-	return &notificationService{}
+// NewNotificationService creates a new notification service instance
+func NewNotificationService(
+	store store.NotificationStorage,
+	delivery DeliveryService,
+	workerLimit int,
+	interval time.Duration,
+	logger *slog.Logger,
+) NotificationService {
+	return &notificationService{
+		store:       store,
+		delivery:    delivery,
+		workerLimit: workerLimit,
+		interval:    interval,
+		l:           logger,
+	}
 }
 
-// SendNotification sends the notification
-func (s *notificationService) SendNotification(ctx context.Context, notification *model.Notification) error {
+// Send sends the notification
+func (s *notificationService) Send(ctx context.Context, n *model.Notification) error {
 	// Simulate sending notification service
-	fmt.Println("Notification: URL", notification.UrlId)
+	s.l.Info("Notification service send called with url ", n.UrlId)
```

Comment:
slog package accepts key,value pairs, fix this log.
Also need to add nil check in pinter `n` before using it.